.github/workflows/pr_ci.yml

```
@@ -47,4 +47,13 @@ jobs:
         path: |
           test/OrchardCore.Tests.Functional/cms-tests/cypress/screenshots
           src/OrchardCore.Cms.Web/App_Data/logs
-  
+    - name: UI Tests
+      if: matrix.os == 'windows-latest'
+      run: |
+        dotnet test -c Release --no-restore --no-build ./test/OrchardCore.Tests.UI/OrchardCore.Tests.UI.csproj
+    - uses: actions/upload-artifact@v2
+      if: matrix.os == 'windows-latest' && failure()
+      with:
+        name: ui-test-failure
+        path: |
+          test/OrchardCore.Tests.UI/bin/Release/net6.0/FailureDumps
```

This can be future-proofed [using a wildcard pattern](https://github.com/actions/upload-artifact/tree/v2#upload-using-a-wildcard-pattern).
```suggestion
          test/OrchardCore.Tests.UI/bin/Release/*/FailureDumps
```

Also I don't think you need `matrix.os == 'windows-latest'` because if nothing is found on the path them the step still successfully completes with only a `Warning: No files were found with the provided path: test/OrchardCore.Tests.UI/bin/Release/*/FailureDumps. No artifacts will be uploaded.` warning message.